cloudops_storage_distribution.go

```diff
@@ -0,0 +1,95 @@
+package cloudops
+
+// CloudStorageDecisionMatrixRow defines an entry in the cloud storage decision matrix.
+type CloudStorageDecisionMatrixRow struct {
+	// IOPS is the desired iops from the underlying cloud storage.
+	IOPS uint32 `json:"iops" yaml:"iops"`
+	// InstanceType is the type of instance on which the cloud storage will
+	// be attached.
+	InstanceType string `json:"instance_type" yaml:"instance_type"`
+	// InstanceMaxDrives is the maximum number of drives that can be attached
+	// to an instance without a performance hit.
+	InstanceMaxDrives uint32 `json:"instance_max_drives" yaml:"instance_max_drives"`
+	// InstanceMinDrives is the minimum number of drives that need to be
+	// attached to an instance to achieve maximum performance.
+	InstanceMinDrives uint32 `json:"instance_min_drives" yaml:"instance_min_drives"`
+	// Region of the instance.
+	Region string `json:"region" yaml:"region"`
+	// MinSize is the minimum size of the drive that needs to be provisioned
+	// to achieve the desired IOPS on the provided instance types.
+	MinSize uint64 `json:"min_size" yaml:"min_size"`
+	// MaxSize is the maximum size of the drive that can be provisioned
+	// without affecting performance on the provided instance type.
+	MaxSize uint64 `json:"max_size" yaml:"max_size"`
+	// Priority for this entry in the decision matrix.
+	Priority string `json:"priority" yaml:"priority"`
+	// ThinProvisioning is set/unset if the cloud provider supports such a
+	// storage device.
+	ThinProvisioning bool `json:"thin_provisioning" yaml:"thin_provisioning"`
+}
+
+// CloudStorageDecisionMatrix is used to determine the optimum cloud storage distribution
+// for a cluster based on user's requirement specified through CloudStorageUserRequirement
+type CloudStorageDecisionMatrix struct {
+	// Rows of the decision matrix
+	Rows []*CloudStorageDecisionMatrixRow `json:"rows" yaml:"rows"`
+}
+
+// CloudStorageUserRequirement is the user provided storage requirement for the cluster.It
+// dictates the min and maximum capacity of a storage type that can be provisioned
+// which provides the desired IOPS. If there is a requirement for two different
+// drive types then multiple CloudUserSpecs need to be provided to the Cloud
+// Drive Management library
```

Comment:
// CloudStorageUserRequirement is the user provided storage requirement for the cluster. This specifies desired capacity thresholds and desired IOPS